src/components/views/elements/BlurhashPlaceholder.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ Copyright 2020 The Matrix.org Foundation C.I.C.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import React from 'react';
+import {decode} from "blurhash";
+
+interface IProps {
+    blurhash: string;
+    width: number;
+    height: number;
+}
+
+export default class BlurhashPlaceholder extends React.PureComponent<IProps> {
+    private canvas: React.RefObject<HTMLCanvasElement> = React.createRef();
+
+    public componentDidMount() {
+        this.draw();
+    }
+
+    public componentDidUpdate() {
+        this.draw();
+    }
+
+    private draw() {
+        if (!this.canvas.current) return;
+
+        try {
+            const {width, height} = this.props;
+
+            const pixels = decode(this.props.blurhash, Math.ceil(width), Math.ceil(height));
+            const ctx = this.canvas.current.getContext("2d");
+            const imgData = ctx.createImageData(width, height);
+            imgData.data.set(pixels);
+            ctx.putImageData(imgData, 0, 0);
+        } catch (e) {
+            console.error("Error rendering blurhash: ", e);
+        }
+    }
+
+    public render() {
+        return <canvas height={this.props.height} width={this.props.width} ref={this.canvas} />;
```

Comment:
given that this uses canvases, what's the failure mode with resist fingerprinting enabled?

Reply:
A big empty div, like we have today.

Reply:
Are you sure it won't cause https://github.com/vector-im/element-web/issues/14433 ?

Reply:
in my testing it didn't.

Reply:
I think fundamentally it can depending on Firefox weirdness :(

Reply:
how do we avoid it? If it's unavoidable, ohwell imo. The MSC doesn't seem to want to support an easy way for us to shove this into an `<img>` and be done with it.

Reply:
perhaps test if the canvases are spitting out sad things and fall back to old behaviour explicitly

Reply:
test how? We already wrap the entire draw function in a try/catch.

Reply:
Test with known value with whether it gives you a stripy thing or the thing you expect

Reply:
That sounds like a browser bug and not something we should be testing for. It's generally assumed that feeding it an image will result in that image being shown - not something we have to make sure the browser is compliant with our request.

I don't think we need to do anything special here.

Reply:
For cases where you use the canvas in response to user input, Firefox (in resist fingerprinting mode) will prompt the user for permission, but for cases like this where there's no user input, you'll silently get the randomized / stripy data.

Of course, it's awkward to test for people in resist fingerprinting mode, since that's sort of the whole point of it. I guess my initial reaction for this feature is that it's a temporary placeholder, so it's perhaps okay to try like this, and we could adjust based on feedback.